File: src/main/java/com/cacodev/shalom/features/member/dto/MemberTypeCreateRequest.java
package com.cacodev.shalom.features.member.dto;

import com.cacodev.shalom.features.member.domain.MemberRole;

public record MemberTypeCreateRequest(
        String name,
        String description,
        double membershipFee,
        int membershipDurationMonths,
        boolean hasVotingRights,
        MemberRole memberRole,
        String privileges
) {}

File: cacodev-ui/src/app/models/MemberTypeCreateRequest.ts
import { MemberRole } from "../enums/memberRole";

export interface MemberTypeCreateRequest {
    name: string;
    description: string;
    membershipFee: number;
    membershipDurationMonths: number;
    hasVotingRights: boolean;
    memberRole: MemberRole;
    privileges: string;
}